gateway/src/service.rs

```diff
@@ -1071,7 +1155,15 @@ pub mod tests {
     #[tokio::test]
     async fn service_create_find_stop_delete_project() -> anyhow::Result<()> {
         let world = World::new().await;
-        let svc = Arc::new(GatewayService::init(world.args(), world.pool(), "".into()).await);
+        let svc = Arc::new(
+            GatewayService::init(
+                world.args(),
+                world.pool(),
+                "".into(),
+                DockerStatsSource::Bollard,
```

Comment:
Hmm, rather than passing in a new arg and defaulting to Bollard, should we do the check we do in gateway main in `GatewayService::init`, and use that to decide the stats source? This way, tests ran locally on Linux and in CI (also Linux) will use the same source as our deployments.

Reply:
Sure, we can do that! I am still getting familiar with Rust and its error handling. I am currently searching for a solution, but do you happen to know where in the project I could find some inspiration on how to throw an error if Bollard is the source type and the environment is staging/production? Or should I consider changing the return type of init to a Result, then handle the error the same as before on `start` in `gateway/main` with a match or something similar?

Reply:
Oh, sorry, I didn't see this comment until now. You handled it well, though!